src/adapter/src/catalog/storage.rs

```diff
@@ -329,6 +329,16 @@ async fn migrate(
             txn.cluster_replicas.update(|_k, v| Some(v.clone()))?;
             Ok(())
         },
+        // Remove the materialize role from existing deployments
+        //
+        // Introduced in v0.45.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.46.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            txn.roles
+                .delete(|_role_key, role_value| &role_value.name == "materialize");
```

Comment:
Also need to add an audit log event dropping the role probably? Super annoying but could otherwise make the audit log slightly incorrect.

Reply:
Good catch. I just added this.

Reply:
Hmm, actually I think my implementation is wrong. I'll have to think of a better way.

Reply:
Ok, I just pushed a correct implementation.